From 2e5a739f8526aee2fe290d75cf65f27d7168c6e1 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.uw@renesas.com>
Date: Wed, 27 Nov 2019 11:26:15 +0700
Subject: [PATCH 3/4] plat: rcar: M3: qos_init_m3_v11: prevent DRAM Split set
 when ECC enabled

When ECC enabled, DRAM Split is not supported.
This commit prevent user to enable DRAM Split with ECC

Signed-off-by: Binh Nguyen <binh.nguyen.uw@renesas.com>
---
 plat/renesas/rcar/qos/M3/qos_init_m3_v11.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plat/renesas/rcar/qos/M3/qos_init_m3_v11.c b/plat/renesas/rcar/qos/M3/qos_init_m3_v11.c
index 8af5291..55237b8 100644
--- a/plat/renesas/rcar/qos/M3/qos_init_m3_v11.c
+++ b/plat/renesas/rcar/qos/M3/qos_init_m3_v11.c
@@ -134,6 +134,9 @@ void qos_init_m3_v11(void)
  #endif
 #elif (RCAR_DRAM_SPLIT == RCAR_DRAM_SPLIT_2CH) || \
 	(RCAR_DRAM_SPLIT == RCAR_DRAM_SPLIT_AUTO)
+#if RZG_DRAM_HIHOPE_RZG2M_ECC
+	NOTICE("BL2: DRAM Split not supported when ECC enabled\n");
+#else
 	NOTICE("BL2: DRAM Split is 2ch\n");
 	io_write_32(AXI_ADSPLCR0, 0x00000000U);
 	io_write_32(AXI_ADSPLCR1, ADSPLCR0_ADRMODE_DEFAULT
@@ -142,6 +145,7 @@ void qos_init_m3_v11(void)
 				  | ADSPLCR0_SWP);
 	io_write_32(AXI_ADSPLCR2, 0x00001004U);
 	io_write_32(AXI_ADSPLCR3, 0x00000000U);
+#endif
 #else
 	NOTICE("BL2: DRAM Split is OFF\n");
 #endif
-- 
2.7.4

